pkg/service/session_stream.go

```diff
@@ -222,6 +231,10 @@ func (s *session) handleRawStream(id string, data map[string][]byte, stream tran
 		// Close the stream with an error.
 		// Ignore, whether this succeeds or not.
 		_ = ts.closeWithErr(sErr)
+	} else {
+		// Handler exited gracefully, close the stream now.
+		// Ignore, whether this succeeds or not.
+		_ = stream.Close()
```

Comment:
Should we not add a defer stream.Close() to the top of this function body and always ensure, that the stream is closed? I must dig deeper into the code base to get a full overview. Maybe you got some hints?

Reply:
Was also my first idea, but the problem is line 245, where the hook is called that the stream is closed.